lib/cli/src/Cardano/CLI.hs

```diff
@@ -1576,26 +1599,87 @@ mkScribeId :: LogOutput -> ScribeId
 mkScribeId (LogToStdout _) = "StdoutSK::text"
 mkScribeId (LogToFile file _) = T.pack $ "FileSK::" <> file
 
+getPrometheusURL :: IO URI
+getPrometheusURL = do
+    prometheus_port <- fromMaybe 13798 . (>>= readMaybe @Int)
+        <$> lookupEnv "CARDANO_WALLET_PROMETHEUS_PORT"
+    pure $ URI {
+          uriScheme = "http"
+        , uriAuthority = Just URIAuth {
+              uriUserInfo = ""
+            , uriRegName = "127.0.0.1"
+            , uriPort = T.unpack (toText prometheus_port)
+        }
+        , uriPath = ""
+        , uriQuery = ""
+        , uriFragment = ""
+        }
+
+getEKGURL :: IO URI
+getEKGURL = do
+    ekg_port <- fromMaybe 13788 . (>>= readMaybe @Int)
+        <$> lookupEnv "CARDANO_WALLET_EKG_PORT"
+    pure $ URI {
+          uriScheme = "http"
+        , uriAuthority = Just URIAuth {
+              uriUserInfo = ""
+            , uriRegName = "127.0.0.1"
+            , uriPort = T.unpack (toText ekg_port)
+            }
+        , uriPath = ""
+        , uriQuery = ""
+        , uriFragment = ""
+        }
 
 -- | Initialize logging at the specified minimum 'Severity' level.
 initTracer
     :: LoggerName
     -> [LogOutput]
     -> IO (Switchboard Text, (CM.Configuration, Trace IO Text))
 initTracer loggerName outputs = do
+    URI { uriAuthority = Just URIAuth {
+            uriRegName = prometheus_host, uriPort = prometheus_port
+        } } <- getPrometheusURL
+    URI { uriAuthority = Just URIAuth {
+            uriRegName = ekg_host, uriPort = ekg_port
+        } } <- getEKGURL
     cfg <- do
         c <- defaultConfigStdout
-        CM.setSetupBackends c [CM.KatipBK, CM.AggregationBK]
+        CM.setSetupBackends c [CM.KatipBK, CM.AggregationBK, CM.EKGViewBK, CM.EditorBK]
+        CM.setDefaultBackends c [CM.KatipBK, CM.EKGViewBK]
         CM.setSetupScribes c $ map mkScribe outputs
         CM.setDefaultScribes c $ map mkScribeId outputs
+        CM.setEKGBindAddr c $ Just (Endpoint (ekg_host, read ekg_port))
```

Comment:
So you still enable EKG for both the wallet and the test tracer? How does that work?

Is it really the job of `initTracer` to read the `CARDANO_WALLET_EKG_PORT` env var? `initTracer` is also called by DB tests through `withLogging`.

Why complicate with `URI` types + parsing and formatting instead of using something more like `(String, Port)` or just `Port`, or `Int`?

How about _either_:
1. Take  two `Maybe Port` as arguments to `initTracer`. Could be put together with `outputs` inside a record if it seems better — `{ outputs, ekgPort, prometheusPort }`
2. Add `get{EKG,Prometheus}Port}FromEnv :: IO Port` to [Shelley.Launch](https://github.com/input-output-hk/cardano-wallet/blob/102fdb9e0052a76afee461f019969596c6efcf2f/lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs#L366-L376)

or 

Setup EKG and Prometheus using `setupEkgAndPrometheus :: Maybe Port -> Maybe Port -> CM.Configuration -> IO ()`, and calling from the integration main module, but I'm guessing you tried tried and it doesn't work?

Reply:
> So you still enable EKG for both the wallet and the test tracer? How does that work?

Nope, if you don't call `EKG.plugin cfg tr sb >>= loadPlugin sb`, then nothing will be exposed.

> Is it really the job of initTracer to read the CARDANO_WALLET_EKG_PORT env var? initTracer is also called by DB tests through withLogging.

I don't understand your concern here. The DB tests don't initialise the ekg plugin, but they are free to do so, if they wanted (e.g. to test logging, if that would be a thing).

> Why complicate with URI types + parsing and formatting instead of using something more like (String, Port) or just Port, or Int?

I don't have a strong opinion on this. So either is fine, I guess. Also feel free to provide a PR if you have detailed ideas on how to design this.

Reply:
I changed this quite a bit now